wicker/core/column_files.py

```diff
@@ -178,16 +183,49 @@ def _write_column(self, column_name: str) -> None:
         self.write_buffers[column_name] = self._get_new_buffer()
 
 
-class ColumnBytesFileCache:
+class ColumnBytesFileReader(abc.ABC):
+    """A reader class for column bytes files to inherit upon for caches and cacheless"""
+
+    def __init__(
+        self,
+        column_bytes_root_path: str,
+    ) -> None:
+        super().__init__()
+        self._column_bytes_root_path = column_bytes_root_path
+
+    def read(self, column_bytes_file_info: ColumnBytesFileLocationV1) -> bytes:
+        """Read data from column bytes file.
+
+        :param column_bytes_file_info: Location object to column bytes file on data store
+        :type column_bytes_file_info: ColumnBytesFileLocationV1
+        :return: bytes read from the column bytes file
+        """
+        column_concatenated_bytes_file_path = os.path.join(
+            self._column_bytes_root_path, str(column_bytes_file_info.file_id)
+        )
+
+        with open(column_concatenated_bytes_file_path, "rb") as f:
+            f.seek(column_bytes_file_info.byte_offset)
+            return f.read(column_bytes_file_info.data_size)
+
```

Comment:
@pickles-bread-and-butter if you want to be super slick, add this function to the class:
```
def read_column_bytes_file(self, column_bytes_file_info: ColumnBytesFileLocationV1, column_bytes_file_path: str) -> bytes:
    with open(column_concatenated_bytes_file_path, "rb") as f:
            f.seek(column_bytes_file_info.byte_offset)
            return f.read(column_bytes_file_info.data_size)
```
and call it from both this `read` function and also the `ColumnBytesFileCache` class `read` function. This is a micro-optimization so it is ok to just mark this comment as resolved if you don't like it.

Reply:
I like this, thanks for the suggestion!